apps/src/code-studio/pd/professional_learning/courses/SelfPacedPLCatalog.tsx

```diff
@@ -0,0 +1,148 @@
+import React, {useEffect, useState} from 'react';
+import {Provider} from 'react-redux';
+
+import SelfPacedPLCatalogCard from '@cdo/apps/code-studio/pd/professional_learning/courses/SelfPacedPLCatalogCard';
+import {updateQueryParam} from '@cdo/apps/code-studio/utils';
+import {
+  filterByDuration,
+  filterByGradeLevel,
+  filterByMarketingInitiative,
+  filterByTopic,
+} from '@cdo/apps/templates/courseOfferings/filters/helpers';
+import NoMatchingSearchResultsFound from '@cdo/apps/templates/courseOfferings/noMatchingSearchResultsFound/NoMathcingSearchResultsFound';
+import {CourseOffering} from '@cdo/apps/templates/courseOfferings/types';
+// import CurriculumCatalog from '@cdo/apps/templates/curriculumCatalog/CurriculumCatalog';
+import teacherSections from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import NoMatchingCoursesImage from '@cdo/static/professional-learning/courses/no-curriculum-assigned-empty-state-illustration.png';
+import PLCatalogHeroBannerImage from '@cdo/static/professional-learning/courses/selfPacedPLCatalog-HeroBanner-illustration.png';
+
+import {
+  getEmptyFilters,
+  getInitialFilterStates,
+  SELF_PACED_PL_CATALOG_FILTERS,
+} from './helpers';
+import SelfPacedPLCatalogFilters from './SelfPacedPLCatalogFilters';
+
+// eslint-disable-next-line import/no-duplicates
+import style from './selfPacedPLCatalog.module.scss';
+// eslint-disable-next-line no-duplicate-imports,import/no-duplicates
+import moduleStyles from './selfPacedPLCatalog.module.scss';
+
+const SelfPacedPLCatalog: React.FunctionComponent<{
+  selfPacedPLCourseOfferings: CourseOffering[];
+}> = ({selfPacedPLCourseOfferings}) => {
+  const [filteredCourses, setFilteredCourses] = useState(
+    selfPacedPLCourseOfferings
+  );
+
+  const [appliedFilters, setAppliedFilters] = useState(
+    getInitialFilterStates()
+  );
+
+  // Updates the filtered courses based on the applied filters.
+  useEffect(() => {
+    const newlyFilteredCourses = selfPacedPLCourseOfferings.filter(
+      course =>
+        filterByGradeLevel(course, appliedFilters.grade) &&
+        filterByTopic(course, appliedFilters.topic) &&
+        filterByMarketingInitiative(
+          course,
+          appliedFilters.marketingInitiative
+        ) &&
+        filterByDuration(course, appliedFilters.duration)
+    );
+
+    setFilteredCourses(newlyFilteredCourses);
+  }, [
+    selfPacedPLCourseOfferings,
+    appliedFilters.grade,
+    appliedFilters.topic,
+    appliedFilters.marketingInitiative,
+    appliedFilters.duration,
+    setFilteredCourses,
+  ]);
+
+  console.log(selfPacedPLCourseOfferings);
+  const [expandedCardKey, setExpandedCardKey] = useState('');
+
+  useEffect(() => {
+    const expandedCardFound = filteredCourses.some(
+      co => expandedCardKey === co['key']
+    );
+
+    if (!expandedCardFound) {
+      setExpandedCardKey('');
+    }
+  }, [expandedCardKey, filteredCourses]);
+
+  // const updateExpandedCardKey = (key: string) => {
```

Comment:
Nit: Can this commented code be removed? Or is it commented out to appease lint rules until you implement it?